catalog/core/catalog-core-validator/src/main/java/ddf/catalog/validation/impl/ReportingMetacardValidatorImpl.java

```diff
@@ -0,0 +1,112 @@
+import ddf.catalog.validation.impl.report.MetacardValidationReportImpl;
+import ddf.catalog.validation.report.MetacardValidationReport;
+import ddf.catalog.validation.violation.ValidationViolation;
+import ddf.catalog.validation.violation.ValidationViolation.Severity;
+
+/**
+ * Default {@link Metacard} validator that validates all of a {@link Metacard}'s attributes using
+ * the {@link AttributeValidator}s registered in the attribute validator registry.
+ */
+public class ReportingMetacardValidatorImpl
+        implements MetacardValidator, ReportingMetacardValidator {
+    private final AttributeValidatorRegistry validatorRegistry;
+
+    public ReportingMetacardValidatorImpl(final AttributeValidatorRegistry validatorRegistry) {
+        this.validatorRegistry = validatorRegistry;
+    }
+
+    private void getMessages(final Set<ValidationViolation> violations, final List<String> warnings,
+            final List<String> errors) {
+        for (final ValidationViolation violation : violations) {
+            if (violation.getSeverity() == Severity.WARNING) {
+                warnings.add(violation.getMessage());
+            } else {
+                errors.add(violation.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void validate(final Metacard metacard) throws ValidationException {
+        final Optional<MetacardValidationReport> reportOptional = validateMetacard(metacard);
+
+        if (reportOptional.isPresent()) {
+            final MetacardValidationReport report = reportOptional.get();
+            final List<String> warnings = new ArrayList<>();
+            final List<String> errors = new ArrayList<>();
+
+            getMessages(report.getAttributeValidationViolations(), warnings, errors);
+            getMessages(report.getMetacardValidationViolations(), warnings, errors);
+
+            final ValidationExceptionImpl exception = new ValidationExceptionImpl();
+            exception.setWarnings(warnings);
+            exception.setErrors(errors);
+
+            throw exception;
+        }
+    }
+
+    @Override
+    public Optional<MetacardValidationReport> validateMetacard(final Metacard metacard) {
+        Preconditions.checkArgument(metacard != null, "The metacard cannot be null.");
+
+        final Set<ValidationViolation> violations = new HashSet<>();
+
+        for (final AttributeDescriptor descriptor : metacard.getMetacardType()
+                .getAttributeDescriptors()) {
+            final String attributeName = descriptor.getName();
+            final Attribute attribute = metacard.getAttribute(attributeName);
+            if (attribute != null) {
+                for (final AttributeValidator validator : validatorRegistry.getValidators(
+                        attributeName)) {
+                    validator.validate(attribute)
+                            .ifPresent(report -> violations.addAll(report.getAttributeValidationViolations()));
```

Comment:
I believe there is a beautiful streaming way to do this. And will attempt as soon as I get to computer unless someone beats me